semgrep-core/tests/OTHER/rules/metavar_cond.yaml

```diff
@@ -5,6 +5,8 @@ rules:
   match:
     and:
     - foo($ARG)
-    - where: re.match(".*bar.*", $ARG)
+    where:
+    - metavariable: $ARG
```

Comment:
I prefer my solution that generalizes where, because metavariable-regex: is redundant
when you support regular expression matching in metavariable-comparison:.
However I can see how it's more backward compatible with the way we were doing it before, so maybe it's
better your way.

Reply:
If I'm understanding you correctly, if you don't like the explicit naming of the metavariable, it could just be:
```
where:
- comparison: re.match(".*bar.*", $ARG)
```

We could also translate old-style `metavariable-regex` to that, and maintain backwards-compatibility.